From c8d0f026c5b0ec30aadd740312a706be8fd51014 Mon Sep 17 00:00:00 2001
From: Haijun Zhang <Haijun.Zhang@freescale.com>
Date: Fri, 9 May 2014 10:59:22 +0800
Subject: [PATCH 087/255] mmc :esdhc: add some properties to esdhc

Add some properties like big-endian, sdhci,auto-cmd12
to support esdhc host on  ls1021a-qds.

Signed-off-by: Haijun Zhang <Haijun.Zhang@freescale.com>
[Kevin: Original patch taken from
LS1021A-SDK-V1.1-ARM-SOURCE-20140815-yocto.iso]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 Documentation/devicetree/bindings/mmc/fsl-esdhc.txt | 3 +++
 arch/arm/boot/dts/ls1021a.dtsi                      | 4 +++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/mmc/fsl-esdhc.txt b/Documentation/devicetree/bindings/mmc/fsl-esdhc.txt
index b7943f3..d181166 100644
--- a/Documentation/devicetree/bindings/mmc/fsl-esdhc.txt
+++ b/Documentation/devicetree/bindings/mmc/fsl-esdhc.txt
@@ -22,6 +22,8 @@ Optional properties:
   - voltage-ranges : two cells are required, first cell specifies minimum
     slot voltage (mV), second cell specifies maximum slot voltage (mV).
     Several ranges could be specified.
+  - big-endian/little-endian: Specifies the endian mode the controller
+    works in.
 
 Example:
 
@@ -33,4 +35,5 @@ sdhci@2e000 {
 	/* Filled in by U-Boot */
 	clock-frequency = <0>;
 	voltage-ranges = <3300 3300>;
+	big-endian;
 };
diff --git a/arch/arm/boot/dts/ls1021a.dtsi b/arch/arm/boot/dts/ls1021a.dtsi
index 1e66312..5082b21 100644
--- a/arch/arm/boot/dts/ls1021a.dtsi
+++ b/arch/arm/boot/dts/ls1021a.dtsi
@@ -115,7 +115,9 @@
 			reg = <0x1560000 0x10000>;
 			interrupts = <0 94 0x04>;
 			clock-frequency = <0>;
-			voltage-ranges = <1800 3300>;
+			voltage-ranges = <1800 1800 3300 3300>;
+			sdhci,auto-cmd12;
+			big-endian;
 			bus-width = <4>;
 			status = "disabled";
 		};
-- 
2.0.2

